lib/wasi-types/src/wasi/wasix_manual.rs

```diff
@@ -324,3 +324,150 @@ impl From<ExitCode> for i32 {
+        #[doc = " edge-triggered monitoring.)"]
+        const EPOLLRDHUP = 1 << 2;
+        #[doc = " There is an exceptional condition on the file descriptor."]
+        #[doc = " See the discussion of POLLPRI in poll(2)."]
+        const EPOLLPRI = 1 << 3;
+        #[doc = " Error condition happened on the associated file"]
+        #[doc = " descriptor.  This event is also reported for the write end"]
+        #[doc = " of a pipe when the read end has been closed."]
+        const EPOLLERR = 1 << 4;
+        #[doc = " Hang up happened on the associated file descriptor."]
+        const EPOLLHUP = 1 << 5;
+        #[doc = " Requests edge-triggered notification for the associated"]
+        #[doc = " file descriptor.  The default behavior for epoll is level-"]
+        #[doc = " triggered.  See epoll(7) for more detailed information"]
+        #[doc = " about edge-triggered and level-triggered notification."]
+        const EPOLLET = 1 << 6;
+        #[doc = " Requests one-shot notification for the associated file"]
+        #[doc = " descriptor.  This means that after an event notified for"]
+        #[doc = " the file descriptor by epoll_wait(2), the file descriptor"]
+        #[doc = " is disabled in the interest list and no other events will"]
+        #[doc = " be reported by the epoll interface.  The user must call"]
+        #[doc = " epoll_ctl() with EPOLL_CTL_MOD to rearm the file"]
+        #[doc = " descriptor with a new event mask."]
+        const EPOLLONESHOT = 1 << 7;
+    }
+}
+
+#[doc = " Epoll operation."]
+#[repr(u32)]
+#[derive(Clone, Copy, PartialEq, Eq, num_enum :: TryFromPrimitive, Hash)]
+pub enum EpollCtl {
+    #[doc = " Add an entry to the interest list of the epoll file descriptor, epfd."]
+    Add,
+    #[doc = " Change the settings associated with fd in the interest list to the new settings specified in event."]
+    Mod,
+    #[doc = " Remove (deregister) the target file descriptor fd from the interest list."]
+    Del,
+}
+impl core::fmt::Debug for EpollCtl {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            EpollCtl::Add => f.debug_tuple("EPOLL_CTL_ADD").finish(),
+            EpollCtl::Mod => f.debug_tuple("EPOLL_CTL_MOD").finish(),
+            EpollCtl::Del => f.debug_tuple("EPOLL_CTL_DEL").finish(),
+        }
+    }
+}
+// TODO: if necessary, must be implemented in wit-bindgen
+unsafe impl ValueType for EpollCtl {
+    #[inline]
+    fn zero_padding_bytes(&self, _bytes: &mut [MaybeUninit<u8>]) {}
+}
+
+unsafe impl wasmer::FromToNativeWasmType for EpollCtl {
+    type Native = i32;
+
+    fn to_native(self) -> Self::Native {
+        self as i32
+    }
+
+    fn from_native(n: Self::Native) -> Self {
+        match n {
+            0 => Self::Add,
+            1 => Self::Mod,
+            2 => Self::Del,
+
+            q => todo!("could not serialize number {q} to enum EpollCtl"),
```

Comment:
I don't think we want a panic here...

Will just have to run the conversion on a plain number in the syscalls, I think?

Reply:
This is not my code, its how the template for it was coded - there are 29 instances of this in the code already on native WASIX deserialization